src/seedu/addressbook/data/person/Person.java

```diff
@@ -83,5 +83,15 @@ public int hashCode() {
     public String toString() {
         return getAsTextShowAll();
     }
+    
+    String getPrintableString(Printable... printables) {
+    	String result = "";
+    	for (int i = 0; i < printables.length; i++) {
+    		result += printables[i];
```

Comment:
you definitely need to use the `.getPrintableString()` method here since Printable does not implement `toString()`. Have you tried running your code?